modules/Channel.md

```diff
@@ -0,0 +1,81 @@
+## Channel Module
+
+A channel is a FIFO data structure used to communicate between different Chapel
+tasks. You can learn more about how go-channels work by following this
+[link](https://gobyexample.com/channels)
+Since the operations on the channel are blocking by default, it enables to send
+values from one task to another in a synchronized manner.
+Here is a simple example in Go.
+
+```
+func main() {
+	// make a channel for string values
+	messages := make(chan string)
+
+	// new task to send a value to the channel
+	// it will wait until other task receives
+	go func() { messages <- "ping" }()
+
+	// receive the sent value
+	msg := <-messages
+	fmt.Println(msg)
+}
+```
+
+The module will have a generic `Channel` class to store the channel buffer and
+support operations on them.
+
+* Here are some functions that the module should have
+
+	* `init(type eltType, size : int)` : Specify the buffer data type and size
+	for creating a Channel.
+
+	* `send(value : eltType)` : This function waits for a space in the buffer
+	and	sends the `value` to the channel.
+
+	* `recv() : eltType` : This function waits until the buffer is empty and
+	returns the value sent to the channel.
+
+	* `close()` : This function is used to close the channel such that no more
+	values will be sent to it.
+
+* `iter these()` : An iterator to receive the values sent to the channel until
+it is closed.
+
+* `select` statements : This statement lets you wait for multiple channel
+operations and executes if anyone of them is ready for communication.
```

Comment:
Let's mention that the Go-style select statement provides atomicity and a randomized ordering. Also, can you describe to what extent it is common (or not) for Go programs to use the select statement?

Reply:
Also Rust's crossbeam package supports the same ideas, right? How do they manage `select` statements?

Reply:
> Also, can you describe to what extent it is common (or not) for Go programs to use the select statement?

`select` is a powerful way to deal with multiple channel operations, but I cannot find any articles related to the use of `select` in Go.

Reply:
Is it common in example code from your favorite Go learning materials?

Reply:
Yeah, select statements are described along with channels in most of the cases.

Reply:
So that's something you can mention in the document.

Reply:
@mppf I have made the changes

Reply:
Thanks - can you add to the key points about select that it's commonly used in Go code working with channels?